distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java

```diff
@@ -81,38 +82,109 @@
 
-    protected static final String INSECURE_SERVICE_ROOT = INSECURE_ROOT + HTTP_PORT + "/services";
+        Port() {
+            this.systemProperty = null;
+        }
 
-    protected static final String REST_PATH = SERVICE_ROOT + "/catalog/";
+        Port(String systemProperty) {
+            this.systemProperty = systemProperty;
+        }
 
-    protected static final String OPENSEARCH_PATH = REST_PATH + "query";
+        String getSystemProperty() {
+            return this.systemProperty;
+        }
 
-    protected static final String CSW_PATH = SERVICE_ROOT + "/csw";
+        String getPort(Integer basePort) {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String OPENSEARCH_SOURCE_ID = "openSearchSource";
+        String getPort() {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String CSW_SOURCE_ID = "cswSource";
+        String getUrl(Integer basePort) {
+            return this.getPort(basePort);
+        }
 
-    protected static final String DDF_HOME_PROPERTY = "ddf.home";
+    }
 
-    protected static String ddfHome;
+    public enum Url {
+        SECURE_ROOT("https://localhost:"),
+        SERVICE_ROOT(SECURE_ROOT, Port.HTTPS_PORT, "/services"),
+        INSECURE_ROOT("http://localhost:"),
+        INSECURE_SERVICE_ROOT(INSECURE_ROOT, Port.HTTP_PORT, "/services"),
+        REST_PATH(SERVICE_ROOT, "/catalog/"),
+        OPENSEARCH_PATH(REST_PATH, "query"),
+        CSW_PATH(SERVICE_ROOT, "/csw"),
+        ADMIN_ALL_SOURCES_PATH(SECURE_ROOT, Port.HTTPS_PORT,
+                "/jolokia/exec/org.codice.ddf.catalog.admin.plugin.AdminSourcePollerServiceBean:service=admin-source-poller-service/allSourceInfo"),
+        ADMIN_STATUS_PATH(SECURE_ROOT, Port.HTTPS_PORT,
+                "/jolokia/exec/org.codice.ddf.catalog.admin.plugin.AdminSourcePollerServiceBean:service=admin-source-poller-service/sourceStatus/");
+
+        private final String root;
+
+        private final Enum[] pieces = new Enum[2];
+
+        Url(String base) {
+            this.root = base;
+        }
+
+        Url(Enum piece, String base) {
+            this.pieces[0] = piece;
+            this.root = base;
+        }
+
+        Url(Enum piece0, Enum piece1, String base) {
+            this.pieces[0] = piece0;
+            this.pieces[1] = piece1;
+            this.root = base;
+        }
+
+        String getUrl(Integer basePort) {
+            String returnUrl = "";
+            for (int i = 0; i < pieces.length; i++) {
+                if (null != pieces[i] && pieces[i] instanceof Port) {
+                    returnUrl += ((Port) pieces[i]).getPort(basePort);
+                } else if (null != pieces[i] && pieces[i] instanceof Url) {
+                    returnUrl += ((Url) pieces[i]).getUrl(basePort);
+                }
+            }
+            returnUrl += root;
+            return returnUrl;
+        }
+
+        String getUrl() {
```

Comment:
This method's impl looks identical to your existing implementation of the overrided version. You could just call the other method, passing in the `basePort` field as a parameter.